src/ResourceManager/Sql/Commands.Sql/Database/Cmdlet/SetAzureSqlDatabase.cs

```diff
@@ -146,26 +188,52 @@ protected override IEnumerable<AzureSqlDatabaseModel> GetEntity()
         protected override IEnumerable<AzureSqlDatabaseModel> ApplyUserInputToModel(IEnumerable<AzureSqlDatabaseModel> model)
         {
             List<Model.AzureSqlDatabaseModel> newEntity = new List<AzureSqlDatabaseModel>();
+            AzureSqlDatabaseModel newDbModel = new AzureSqlDatabaseModel()
+            {
+                ResourceGroupName = ResourceGroupName,
+                ServerName = ServerName,
+                DatabaseName = DatabaseName,
+                MaxSizeBytes = MaxSizeBytes,
+                Tags = TagsConversionHelper.ReadOrFetchTags(this, model.FirstOrDefault().Tags),
+                ElasticPoolName = ElasticPoolName,
+                Location = model.FirstOrDefault().Location,
+                ReadScale = ReadScale,
+                ZoneRedundant =
+                       MyInvocation.BoundParameters.ContainsKey("ZoneRedundant")
```

Comment:
Can you check `ZoneRedundant != null` instead? I like to avoid `MyInvocation.BoundParameters.ContainsKey` where possible